zerver/lib/bugdown/__init__.py

```diff
@@ -144,7 +144,12 @@ def add_a(root, url, link, title=None, desc=None,
     if data_id is not None:
         a.set("data-id", data_id)
     img = markdown.util.etree.SubElement(a, "img")
-    img.set("src", url)
+    thumbnail_path = '/thumbnail'
+    if not url.startswith('/'):
+        thumbnail_path += '/'
```

Comment:
OK; we should make sure that we have Bugdown tests covering the different cases in this function, to protect us from breaking things in a future refactoring.